sky/clouds/azure.py

```diff
@@ -132,23 +140,47 @@ def get_egress_cost(self, num_gigabytes: float):
         cost += 0.0
         return cost
 
+    @classmethod
+    def get_default_instance_type(
+            cls,
+            cpus: Optional[str] = None,
+            memory: Optional[str] = None,
+            disk_tier: Optional[resources_utils.DiskTier] = None
+    ) -> Optional[str]:
+        return service_catalog.get_default_instance_type(cpus=cpus,
+                                                         memory=memory,
+                                                         disk_tier=disk_tier,
+                                                         clouds='azure')
+
     @classmethod
     def get_image_size(cls, image_id: str, region: Optional[str]) -> float:
+        compute_client = azure.get_client('compute', cls.get_project_id())
         if region is None:
             # The region used here is only for where to send the query,
             # not the image location. Azure's image is globally available.
             region = 'eastus'
         is_skypilot_image_tag = False
         if image_id.startswith('skypilot:'):
-            is_skypilot_image_tag = True
             image_id = service_catalog.get_image_id_from_tag(image_id,
                                                              clouds='azure')
-        image_id_splitted = image_id.split(':')
-        if len(image_id_splitted) != 4:
-            with ux_utils.print_exception_no_traceback():
-                raise ValueError(f'Invalid image id: {image_id}. Expected '
-                                 'format: <publisher>:<offer>:<sku>:<version>')
-        publisher, offer, sku, version = image_id_splitted
+            if image_id.startswith('/CommunityGalleries'):
+                # Avoid querying the image size from Azure as
+                # all skypilot custom images have the same size.
+                return _DEFAULT_SKYPILOT_IMAGE_GB
+
+        azure_utils.validate_image_id(image_id)
+
+        # Azure community gallery image passed in by the user.
+        if image_id.startswith('/CommunityGalleries'):
+            _, _, gallery_name, _, image_name = image_id.split('/')
+            try:
+                return azure_utils.get_community_image_size(
+                    compute_client, gallery_name, image_name, region)
+            except exceptions.ResourcesUnavailableError:
+                return 0.0
+
+        # Azure marketplace image
+        publisher, offer, sku, version = image_id.split(':')
         if is_skypilot_image_tag:
             if offer == 'ubuntu-hpc':
                 return _DEFAULT_AZURE_UBUNTU_HPC_IMAGE_GB
```

Comment:
Should we move these lines (also the else below, GH does not allow me to select those lines as well) to L170 as well, i.e. shortcut the checks?